core/ui/prime/components/src/lib/tree/tree.component.ts

```diff
@@ -156,4 +157,19 @@ export class TreeComponent extends InputBaseComponent {
 
 		this.valueChanged();
 	}
-}
+
+	public override setDisabledState(isDisabled: boolean): void {
+		this.disabled = isDisabled;
+		for (const node of this.hierarchicalNodes) {
```

Comment:
If you use this._flatmap there is no need to use recursion

Reply:
for (const node of this._flatmap) {
    node.selectable = !disabled
}

Reply:
Fixed

Reply:
Couldn't fix it this way. I still needed to convert `_flatMap` to `hierarchicalNodes` afterwards, which made a small mess of the treeview. Didn't want to spend the effort to debug.

Reply:
Fixed.